src/components/HomePage/SearchArea/SearchArea.js

```diff
@@ -90,19 +94,29 @@ export const AdvancedSearchController = ({
     }, [onSubmit, searchOffers, searchValue, setUrlSearchValue]);
 
     useEffect(() => {
-        if (queryParams.jobMinDuration && queryParams.jobMaxDuration) {
-            setShowJobDurationSlider(true);
+        // eslint-disable-next-line no-constant-condition
+        if (loadUrlFromFilters) {
+            setUrlFilters(
+                jobMinDuration,
+                jobMaxDuration,
+                fields,
+                technologies,
+                jobType,
+                searchValue,
+            );
+            setLoadUrlFromFilters(false);
+            submitForm(null, false);
+        } else {
+            setShowJobDurationSlider(queryParams.jobMinDuration && queryParams.jobMaxDuration);
             setJobDuration(null, [
                 parseInt(queryParams.jobMinDuration, 10),
                 parseInt(queryParams.jobMaxDuration, 10),
             ]);
-        }
 
-        if (queryParams.jobType) setJobType(queryParams.jobType);
-        if (queryParams.fields) setFields(queryParams.fields);
-        if (queryParams.technologies) setTechs(queryParams.technologies);
+            setJobType(queryParams.jobType);
+            setFields(queryParams.fields || []);
+            setTechs(queryParams.technologies || []);
 
-        if (queryParams.searchValue) {
             setSearchValue(queryParams.searchValue);
         }
```

Comment:
I'm not proud of these particular lines of code, any idea on how to improve this bit?